GOVERNANCE.md

```diff
@@ -0,0 +1,147 @@
+
+* [Ryan Williams](https://github.com/ryan-williams)
+
+The SC membership is revisited every January. SC members who do
+not actively engage with the SC duties are expected to resign. New members are
+added by nomination by a core developer. Nominees should have demonstrated
+long-term, continued commitment to the project and its [mission and values](MISSION_AND_VALUES.md). A
+nomination will result in discussion that cannot take more than a month and
+then admission to the SC by consensus. During that time deadlocked votes of the SC will
+be postponed until the new member has joined and another vote can be held.
+
+The zarr steering council may be contacted at `zarr-steering-council@googlegroups.com`.
+Or via the [@zarr-developers/steering-council](https://github.com/orgs/zarr-developers/teams/steering-council) GitHub team.
+
+# Decision Making Process
+
+Decisions about the future of the project are made through discussion with all
+members of the community. All non-sensitive project management discussion takes
+place on the issue trackers of the https://github.com/zarr-developers repositories.
+Occasionally, sensitive discussion may occur on a private core developer mailing list
+`zarr-core-devs@googlegroups.com` or private chat channel.
+
+TBD: Alternative is via https://github.com/orgs/zarr-developers/teams/core-devs
+
+Decisions should be made in accordance with the [mission and values](MISSION_AND_VALUES.md)
+of the zarr project.
+
+zarr uses a “consensus seeking” process for making decisions. The group
+tries to find a resolution that has no open objections among core developers.
+Core developers are expected to distinguish between fundamental objections to a
+proposal and minor perceived flaws that they can live with, and not hold up the
+decision-making process for the latter.  If no option can be found without
+objections, the decision is escalated to the SC, which will itself use
+consensus seeking to come to a resolution. In the unlikely event that there is
+still a deadlock, the proposal will move forward if it has the support of a
+simple majority of the SC.
+
+Decisions (in addition to adding core developers and SC membership as above)
+are made according to the following rules:
+
+- **Minor documentation changes**, such as typo fixes, or addition / correction of a
+  sentence, require approval by a core developer *and* no disagreement or requested
+  changes by a core developer on the issue or pull request page (lazy
+  consensus). Core developers are expected to give “reasonable time” to others
+  to give their opinion on the pull request if they’re not confident others
+  would agree.
+
+- **Code changes and major documentation changes** require agreement by *one*
+  core developer *and* no disagreement or requested changes by a core developer
+  on the issue or pull-request page (lazy consensus). For all changes of this type,
+  core developers are expected to give “reasonable time” after approval and before
```

Comment:
Would it be better to narrow down *reasonable* time here. I know this is very context-dependent, but maybe it would be good to clarify if we are talking about hours, days weeks or months.

Reply:
Would work for me, @meggart. Anything in particular you'd propose?

Reply:
I am not sure since I am not contributing to the Python codebase myself. A suggestion would be that we require a minimum delay of 2 weeks since opening the PR and 3 days since approval before a pull request gets merged. However feel free to suggest different times, I just think it takes away uncertainty from the core-dev who is merging the PR.

Reply:
Pushed a first approximation for these times.